### PR TITLE
fix: fix evm_to_starknet_address mapping

### LINKED ISSUE #759: bug: `setup_test_state` error evm to starknet mapping
### Current behavior

The storage var for `evm_to_starknet_address` is set for the current contract address, while it should be set for the Kakarot contract.

https://github.com/kkrt-labs/kakarot/blob/e94997389b0b741ef8989c32b427cfa933a3aaee/tests/integration/ef_tests/test_ef_blockchain_tests.py#L44

### Expected behavior
The `evm_to_starknet_address` should be set for Kakarot contract. Update the test utils for the pre-state construction in order to set the storage at the Kakarot address.

### PR DESCRIPTION
**Pull request by Eikix**
Time spent on this PR: 0.1

## Pull request type

Please check the type of change your PR introduces:

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

Resolves #759

### CODE REVIEW

File: tests/integration/ef_tests/test_ef_blockchain_tests.py [hunk 1]
```diff
@@ -41,8 +41,6 @@ async def setup_test_state(
             storage_entries = (
                 (("is_initialized",), 1),
                 (("evm_address",), evm_address),
-                (("evm_to_starknet_address", evm_address), starknet_address),
-            ) + (
```
Comment (greged93) [inline tests/integration/ef_tests/test_ef_blockchain_tests.py:45]: should this + sign be removed as well?

File: tests/integration/ef_tests/test_ef_blockchain_tests.py [hunk 2]
```diff
@@ -67,6 +65,11 @@ async def setup_test_state(
                 starknet.state.state.cache._nonce_writes[starknet_address] = int(
                     account["nonce"], 16
                 )
+                await starknet.state.state.set_storage_at(
```
Comment (greged93) [inline tests/integration/ef_tests/test_ef_blockchain_tests.py:68]: It looks like you are only doing this for eoa's, but I think it needs to be done for all addresses, be it eoa or contract accounts.

File: tests/integration/ef_tests/test_ef_blockchain_tests.py (hunk 1 above)
Comment (Eikix) [reply tests/integration/ef_tests/test_ef_blockchain_tests.py:45]: traumatized by python, i didnt know what it did

Comment (jobez) [reply tests/integration/ef_tests/test_ef_blockchain_tests.py:45]: it basically combines the base tuples (left hand side) with either the necessary eoa or contract tuples!

**Comment by Eikix** [comment]
Checked locally and this seems to work

**Review by greged93 — APPROVED** [review]
lgtm

**Pull request merged**